gateway/modules/eventing/eventing.go

```diff
@@ -108,15 +128,29 @@ func (m *Module) SetConfig(projectID string, eventing *config.EventingConfig) er
 		return errors.New("invalid eventing config provided")
 	}
 
+	fmt.Println("Current p config vs new p config", m.config.ProcessingConfig, eventing.ProcessingConfig)
+	if eventing.ProcessingConfig == 0 {
+		fmt.Println("Provided processing config is 0, default to", utils.DefaultEventProcessingConfig)
+		eventing.ProcessingConfig = utils.DefaultEventProcessingConfig
+	}
+	if m.config.ProcessingConfig != eventing.ProcessingConfig {
+		_ = fmt.Sprintf("Processing config has been changed from (%d) to (%d)", m.config.ProcessingConfig, eventing.ProcessingConfig)
+		m.bufferedEventProcessingRoutineCloser <- struct{}{}
+		m.bufferedEventProcessingChannel = make(chan *model.EventDocument, eventing.ProcessingConfig*3)
```

Comment:
After thinking about this. Lets hard code the parallel worker count. Putting locks on channels isn't a good thing. We have to use locks if this channel size will be variable